jsettlers.main.android/src/main/java/jsettlers/main/android/dialogs/DirectoryPickerDialog.java

```diff
@@ -0,0 +1,140 @@
+package jsettlers.main.android.dialogs;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.os.Environment;
+import android.os.Handler;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ListView;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Stack;
+
+import jsettlers.main.android.resources.scanner.ResourceLocationScanner;
+
+/**
+ * Created by tingl on 27/05/2016.
+ */
+public class DirectoryPickerDialog extends DialogFragment {
+
+    private ArrayAdapter<File> directoryAdapter;
+    private Stack<File> directoryStack;
+
+    public  interface Listener {
+        void onDirectorySelected();
+    }
+
+    public static DirectoryPickerDialog newInstance() {
+        DirectoryPickerDialog fragment = new DirectoryPickerDialog();
+        //fragment.setCancelable(false);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        directoryStack = new Stack<>();
+        directoryStack.push(Environment.getExternalStorageDirectory());
```

Comment:
This may not be enough. It may be hard to get the right path here - In my experience every device has it's own way of defining this and there is no common way of getting all paths - even though Android docs tell you otherwise.